src/test/java/bio/terra/service/snapshot/SnapshotConnectedTest.java

```diff
@@ -298,6 +302,48 @@ public void testDeleteRecreateSnapshot() throws Exception {
     connectedOperations.getSnapshotExpectError(summaryModelSequel.getId(), HttpStatus.NOT_FOUND);
   }
 
+  @Test
+  public void testProjectDeleteAfterSnapshotDelete() throws Exception {
+    // create a dataset and load some tabular data
+    DatasetSummaryModel datasetSummary =
+        SnapshotConnectedTestUtils.createTestDataset(
+            connectedOperations, billingProfile, "snapshot-test-dataset.json");
+    SnapshotConnectedTestUtils.loadCsvData(
+        connectedOperations,
+        jsonLoader,
+        storage,
+        testConfig.getIngestbucket(),
+        datasetSummary.getId(),
+        "thetable",
+        "snapshot-test-dataset-data-row-ids.csv");
+
+    // create a snapshot
+    SnapshotRequestModel snapshotRequest =
+        SnapshotConnectedTestUtils.makeSnapshotTestRequest(
+            jsonLoader, datasetSummary, "snapshot-test-snapshot.json");
+    MockHttpServletResponse response = performCreateSnapshot(snapshotRequest, "_dup_");
+    SnapshotSummaryModel summaryModel = validateSnapshotCreated(snapshotRequest, response);
+
+    // retrieve snapshot and store project id
+    SnapshotModel snapshotModel =
+        SnapshotConnectedTestUtils.getTestSnapshot(
+            mvc, objectMapper, summaryModel.getId(), snapshotRequest, datasetSummary);
+    assertNotNull("fetched snapshot successfully after creation", snapshotModel);
+    String googleProjectId = snapshotModel.getDataProject();
+
+    // ensure that google project exists
+    Assert.assertNotNull(googleResourceManagerService.getProject(googleProjectId));
+
+    // delete snapshot
+    connectedOperations.deleteTestSnapshot(snapshotModel.getId());
+    connectedOperations.getSnapshotExpectError(snapshotModel.getId(), HttpStatus.NOT_FOUND);
+
+    // check that google project doesn't exist
+    Assert.assertEquals(
+        LifecycleState.DELETE_REQUESTED.toString(),
+        googleResourceManagerService.getProject(googleProjectId).getLifecycleState());
+  }
+
```

Comment:
I like how you added comments and cleanly separated the different sections - makes it easy to read!